lib/decodeHex.ts

```diff
@@ -0,0 +1,12 @@
+import { BigNumber } from '@ethersproject/bignumber'
+
+export function decodeHex(value: string): string {
+  const isHex = new RegExp(/0[xX][0-9a-fA-F]+/)
+
+  if (isHex.test(value)) {
+    value = value.toLowerCase()
```

Comment:
is this needed?

Reply:
BigNumber doesn't convert values properly if we pass to it e.g "0X..", so pretty much yes, If we want to keep accepting values with "X" instead of "x".

Reply:
BigNumber doesn't convert values properly if we pass to it e.g "0X..", so pretty much yes, If we want to keep accepting values with "X" instead of "x".